tutorials/flappy-bird/step_8/flappy-bird.cpp

```diff
@@ -0,0 +1,660 @@
+#include <antara/gaming/world/world.app.hpp>
+#include <antara/gaming/graphics/component.sprite.hpp>
+#include <antara/gaming/input/virtual.hpp>
+
+// For convenience
+using namespace antara::gaming;
+
+// Constants
+struct flappy_bird_constants {
+    // UI
+    const unsigned long long font_size{32ull};
+
+    // Player
+    const float player_pos_x{400.0f};
+    const float gravity{2000.f};
+    const float jump_force{650.f};
+    const float rotate_speed{100.f};
+    const float max_angle{60.f};
+
+    // Pipes
+    const float gap_height{265.f};
+    const float column_start_distance{700.f};
+    const float column_min{0.2f};
+    const float column_max{0.8f};
+    const float column_thickness{100.f};
+    const float column_distance{400.f};
+    const std::size_t column_count{6};
+    const float pipe_cap_extra_width{10.f};
+    const float pipe_cap_height{50.f};
+    const graphics::color pipe_color{92, 181, 61};
+    const graphics::outline_color pipe_outline_color{2.0f, graphics::color{76, 47, 61}};
+    const float scroll_speed{200.f};
+    const std::string player_image_name{"player.png"};
+
+    // Background
+    const float ground_thickness{100.0f};
+    const float grass_thickness{20.0f};
+    const graphics::color background_color{82, 189, 199};
+    const graphics::color ground_color{220, 209, 143};
+    const graphics::color grass_color{132, 227, 90};
+    const graphics::outline_color grass_outline_color{2.0f, graphics::color{76, 47, 61}};
+};
+
+// Random number generator
+namespace {
+    std::random_device rd;  // Will be used to obtain a seed for the random number engine
+    std::mt19937 gen(rd()); // Standard mersenne_twister_engine seeded with rd()
+    float random_float(float lower, float higher) {
+        std::uniform_real_distribution<float> dist(lower, higher);
+        return dist(gen);
+    }
+}
+
+// A Flappy Bird column which has two pipes
+struct pipe {
+    entt::entity body{entt::null};
+    entt::entity cap{entt::null};
+
+    // Destroy pipe
+    void destroy(entt::registry &registry) {
+        registry.destroy(body);
+        registry.destroy(cap);
+    }
+};
+
+// Column is made of two pipes
+struct column {
+    // Entities representing the Flappy Bird pipes
+    pipe top_pipe{entt::null};
+    pipe bottom_pipe{entt::null};
+
+    // Is score taken from this column
+    bool scored{false};
+
+    // Destroy pipes and this column
+    void destroy(entt::registry &registry, entt::entity entity) {
+        top_pipe.destroy(registry);
+        bottom_pipe.destroy(registry);
+        registry.destroy(entity);
+    }
+};
+
+// Score struct, has current value, max record, and the UI text
+struct score {
+    int value;
+    int max_score;
+    entt::entity text;
+};
+
+// Logic functions
+namespace {
+    // Create the UI string
+    std::string score_ui_text(int score = 0, int best_score = 0) {
+        return std::string("Score: ") + std::to_string(score) +
```

Comment:
Here I would prefer to use string literals: `"Score: "_s` over `std::string("Score: ")`